accounts-db/src/accounts_db.rs

```diff
@@ -8291,16 +8383,19 @@ impl AccountsDb {
         let mut num_accounts_rent_paying = 0;
         let mut amount_to_top_off_rent = 0;
         let mut stored_size_alive = 0;
+        let mut zero_pubkeys = vec![];
 
         let (dirty_pubkeys, insert_time_us, mut generate_index_results) = {
             let mut items_local = Vec::default();
             storage.accounts.scan_index(|info| {
                 stored_size_alive += info.stored_size_aligned;
                 if info.index_info.lamports > 0 {
                     accounts_data_len += info.index_info.data_len;
+                    zero_pubkeys.push(info.index_info.pubkey);
```

Comment:
is this missing an `else`? we know these accounts are NOT 0 lamports!

Reply:
yeah. fixed. 

It looks like that we may have a bug for `accounts_data_len`. When `accounts_data_len` was populated here, we exclude zero lamport accounts. But when we visit it in dups, we don't exclude zero lamports.

Here is a follow up PR to fix the issue. https://github.com/anza-xyz/agave/pull/3276